crates/bevy_ecs/src/error/command_handling.rs

```diff
@@ -0,0 +1,110 @@
+use core::{any::type_name, fmt};
+
+use crate::{
+    entity::Entity,
+    error::panic,
+    system::{entity_command::EntityCommandError, Command, EntityCommand},
+    world::{error::EntityMutableFetchError, World},
+};
+
+use super::{BevyError, ErrorContext, GLOBAL_ERROR_HANDLER};
+
+/// Takes a [`Command`] that returns a Result and uses a given error handler function to convert it into
+/// a [`Command`] that internally handles an error if it occurs and returns `()`.
+pub trait HandleError<Out = ()> {
```

Comment:
I'm sort of confused why this is a generic with an empty default. The docs don't make it clear to me either. Can you help me out here?

Reply:
This was moved unchanged in this PR. It was introduced in 7f622096463e35e3ef38c2c59a68fa4c5e1f0f367632eaf1e9c4779fcc72d74eR69.

Reply:
After reading through the rest of the implementation it makes sense.

Reply:
This makes more sense to me now that I've read the rest of the code.